armi/utils/plotting.py

```diff
@@ -1047,3 +1046,237 @@ def makeHistogram(x, y):
         yHistogram[upper] = y[i]
     xHistogram[0] = x[0] / 2.0
     return xHistogram, yHistogram
+
+
+def _makeBlockPinPatches(block, temp):
+    """Return lists of block component patches and corresponding data and names (which relates to material
+    of the component for later plot-coloring/legend) for a single block.
+
+
+    Takes in a block that must have a spatialGrid attached as well as a variable
+    which signifies whether the dimensions of the components are at hot or cold temps.
+    When cold is set to true, you would get the BOL cold temp dimensions.
+
+    Parameters
+    ----------
+    block : block
+
+    temp : boolean
+        cold = true, hot = false
+
+    Return
+    ------
+    patches : list
+        list of patches for block components
+
+    data : list
+        list of the materials these components are made of
+
+    name : list
+        list of the names of these components
+    """
+    patches = []
+    data = []
+    name = []
+    if isinstance(block.spatialGrid, grids.HexGrid):
+        nSides = 6
+        smallestInnerPitch = block.getPitch()
+
+    elif isinstance(block.spatialGrid, grids.ThetaRZGrid):
+        raise ValueError(
+            "This plot function is not currently supported for ThetaRZGrid grids."
+        )
+    else:
+        nSides = 4
+        smallestInnerPitch = block.getPitch()[0]
+        if block.getPitch()[0] != block.getPitch()[1]:
+            raise ValueError("Only works for blocks with equal length and width.")
+
+    sortedComp = []
+
+    # Finds the smallest innerPitch(for HexBlock), or widthInner(for Cartesian)
+    #  which will be used later for placing the DerivedShape Element.
+    for component in block:
+        sortedComp.append(component)
+        if nSides == 6:
+            if "ip" in component.DIMENSION_NAMES:
+                if component.getDimension("ip", cold=temp) < smallestInnerPitch:
+                    smallestInnerPitch = component.getDimension("ip", cold=temp)
+        else:
+            if "widthInner" in component.DIMENSION_NAMES:
+                if (
+                    component.getDimension("widthInner", cold=temp) < smallestInnerPitch
+                    and component.getDimension("widthInner", cold=temp) != 0.0
+                ):
+                    smallestInnerPitch = component.getDimension("widthInner", cold=temp)
+
+    for components in sorted(sortedComp):
```

Comment:
Draw the DerivedShape up here, by using `b.getPitch(returnComp=True)`, then draw that component using the Material of the DerivedShape. You will probably have to somehow extract the DerivedShape from the collection of `sortedComp`onents

Reply:
You probably wont have to do the smallestInnerPitch search anymore as well.